### ISSUE TITLE
Not able to use ajax result to cancel onItemInserting()

### ISSUE THREAD

**Issue by JeffMatthews2** [tabalinas/jsgrid#299]
Example from documentation:
```
onItemInserting: function(args) {
        // cancel insertion of the item with empty 'name' field
        if(args.item.name === "") {
            args.cancel = true;
            alert("Specify the name of the item!");
        }
    }
```

What I want to do:
```
function checkNewEvent(myEvent) {
        return $.ajax({
            type: "POST",
            url: "/Event/GetEventByDateTime",
            data: { getEvent: myEvent },
        });
    }

<JsGrid stuff here>

onItemInserting: function (args) {

            // Check for duplicates in database.
            var myEvent = args.item;

            var isDuplicate = checkNewEvent(myEvent).done(function (r)
            {
                if (r.Id)
                {
                    // Event with same date was found.
                    alert("There is already an event scheduled for this date.");
                    args.cancel = true;                    
                }
            }).fail(function (x)
            {
                alert("request to server failed"); // Tell the user something bad happened
                args.cancel = true;
            });
```

The ajax works properly.  However, the problem is that onItemInserting() completes and passes control to inserting BEFORE the ajax completes.  I am struggling with how to handle async here.  My attempts to set args.cancel to true are useless.  I have seen admonitions to avoid forcing synchronous request, and so I don't want to do that if I can avoid it.

**Comment by tabalinas** [comment]
For now async callbacks are not supported. You can check duplicated not with the separate ajax call, but in the insertion call before actual insertion. If duplicate will be found you just return according HTTP status with an error and handle error case in [`onError`](http://js-grid.com/docs/#onerror) callback showing alert or any other way.

**Comment by JeffMatthews2** [comment]
It sounds odd to send an HTTP error when a duplicate is found.  I am still honing my skills to some degree and would have to experiment with the idea.  However, if a valid HTTPRequest is made and processed with no errors, forcing it to return an error would seem to obfuscate the logic and possibly confuse the response with things like a down server, no internet connection, etc.  I would want to know the request was successful and found a matching record.

Do you have an example on your end for checking the database for duplicates and cancelling an insert?

**Comment by tabalinas** [comment]
No, it's totally normal, your REST API returns an error in exceptional situation. Thus if user tries to add a duplicate, the insertion fails and APi should return e.g. 400 Bad Request. See this SO answer http://stackoverflow.com/questions/3290182/rest-http-status-codes-for-failed-validation-or-invalid-duplicate

**Comment by JeffMatthews2** [comment]
Thanks for your suggestion and the link.  Something more yet to study and learn....

**Comment by AlexPerotti** [comment]
Hi Jeff,

it seems to me that you actually don't want an async ajax call.

Have you tried to set async = false on your ajax settings?

**Comment by DanielKingCA** [comment]
onItemInserting: function (args) {

                    var gridData = $("#myGrid").jsGrid("option", "data");
                    for (var i = 0; i < gridData.length; i++)
                    {
                        if (args.item.myField == gridData[i].myField)
                        {
                            args.cancel = true;
                            alert(args.item.myField + ' already exists.  Please enter a unique value.');
                        }
                    }
                },
